### PR TITLE
[KIECLOUD-122] - Enable JMS for RHDM and RHPAM images.

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: Filippe Spolti <fspolti@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[RHPAM-XYZ] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

File: templates/rhpam73-prod-immutable-kieserver-amq.yaml [hunk 1]
```diff
@@ -0,0 +1,1214 @@
...
+  required: true
+- displayName: Database Volume Capacity
+  description: Size of persistent storage for the database volume.
+  name: DB_VOLUME_CAPACITY
+  value: 1Gi
+  required: true
+## PostgreSQL database parameters END
+- displayName: KIE Server Mode
+  description: "The KIE Server mode. Valid values are 'development' or 'production'. (Sets the org.kie.server.mode system property)."
+  name: KIE_SERVER_MODE
```
Comment (errantepiphany) [inline templates/rhpam73-prod-immutable-kieserver-amq.yaml]: KIE_SERVER_MODE is no longer a configurable parameter, although we do set it to a hardcoded value in the DeploymentConfig in this particular immutable template. See rhpam73-prod-immutable-kieserver.yaml for an example.

File: templates/rhpam73-prod-immutable-kieserver-amq.yaml [hunk 2]
```diff
@@ -0,0 +1,1214 @@
...
+          - name: ping
+            containerPort: 8888
+            protocol: TCP
+          env:
+          - name: KIE_ADMIN_USER
+            value: "${KIE_ADMIN_USER}"
+          - name: KIE_ADMIN_PWD
+            value: "${KIE_ADMIN_PWD}"
+          - name: KIE_SERVER_MODE
+            value: "${KIE_SERVER_MODE}"
```
Comment (errantepiphany) [inline templates/rhpam73-prod-immutable-kieserver-amq.yaml]: KIE_SERVER_MODE should be hardcoded to DEVELPOMENT in this template. See rhpam73-prod-immutable-kieserver.yaml for an example.

File: templates/rhpam73-prod-immutable-kieserver-amq.yaml [hunk 3]
```diff
@@ -0,0 +1,1214 @@
+---
```
Comment (errantepiphany) [inline templates/rhpam73-prod-immutable-kieserver-amq.yaml]: For the master branch, this file name needs to have an rhpam74 prefix, not rhpam73.

Comment (errantepiphany) [reply templates/rhpam73-prod-immutable-kieserver-amq.yaml]: Also all "73 and 7.3" references in the file need to get updated as well.

File: templates/rhpam73-prod-immutable-kieserver-amq.yaml (hunk 1 above)
Comment (spolti) [reply templates/rhpam73-prod-immutable-kieserver-amq.yaml]: ok.

File: templates/rhpam73-prod-immutable-kieserver-amq.yaml (hunk 2 above)
Comment (spolti) [reply templates/rhpam73-prod-immutable-kieserver-amq.yaml]: updated.

File: templates/rhpam73-prod-immutable-kieserver-amq.yaml (hunk 3 above)
Comment (spolti) [reply templates/rhpam73-prod-immutable-kieserver-amq.yaml]: Old PR, changes made before the version update :)

**Comment by spolti** [comment]
Good catch, didn't see those lines.

**Pull request merged**